internal/cloud/realm/error.go

```diff
@@ -0,0 +1,36 @@
+package realm
+
+import (
+	"bytes"
+	"encoding/json"
+	"net/http"
+)
+
+// ServerError is a Realm server error
+type ServerError struct {
+	Code    string `json:"error_code"`
+	Message string `json:"error"`
+}
+
+func (se ServerError) Error() string {
+	return se.Message
+}
+
+// UnmarshalServerError attempts to unmarshal a server error from the *http.Response
+func UnmarshalServerError(res *http.Response) error {
+	buf := new(bytes.Buffer)
+	if _, err := buf.ReadFrom(res.Body); err != nil {
+		return err
+	}
+
+	payload := buf.String()
+	if payload == "" {
+		return ServerError{Message: res.Status}
```

Comment:
should this be `Code`?

Reply:
`Code` is meant to be the `baas` concept of "error code" (e.g. `"InvalidSession"`).  `Message` is ultimately what gets propagated out as the error string, so I want `Message` set here with the response's status text (should no payload be present).

Note: we're not capturing the response's _status code_ at all in this unmarshal process (i.e. whether it's a `401` or `500` or w/e)